zng/resolver/mapper.go

```diff
@@ -53,6 +53,18 @@ func (m *Mapper) Translate(foreign zng.Type) (zng.Type, error) {
 
 func (m *Mapper) EnterType(td int, typ zng.Type) {
 	m.mu.Lock()
-	m.Slice.Enter(td, typ)
+	if td >= len(m.types) {
+		new := make([]zng.Type, td+1, td*2)
+		copy(new, m.types)
+		m.types = new
+	}
```

Comment:
Nit: No need to copy if `types` has sufficient capacity.
```suggestion
	if td >= cap(m.types) {
		new := make([]zng.Type, td+1, td*2)
		copy(new, m.types)
		m.types = new	
	} else if td >= len(m.types) {
		m.types = t.type[:td+1]
	}
```